### PR TITLE
AN-6641: Settings Support Screen

### PR DESCRIPTION
**Pull request by BradleyWire**
## What's new in this PR?

JIRA: https://wearezeta.atlassian.net/browse/AN-6641

Refactored the settings support screen to rely less on strings.xml and made it testable 

#### APK
[Download build #1166](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1166/artifact/build/artifact/wire-dev-PR2590-1166.apk)
[Download build #1186](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1186/artifact/build/artifact/wire-dev-PR2590-1186.apk)

### CODE REVIEW

File: app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt [hunk 1]
```diff
@@ -5,20 +5,41 @@ import android.view.LayoutInflater
...
+
+        settingsSupportSupportWebsiteButton.setOnClickListener {
+            settingsSupportViewModel.onSupportWebsiteClicked()
+        }
+        settingsSupportContactButton.setOnClickListener {
+            settingsSupportViewModel.onSupportContactClicked()
+        }
+    }
+
+    private fun initViewModel() {
```
Comment (android10) [inline app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt]: I'm not a big fan of this kind of naming telling us about implementation details (the fact we are using an Android ViewModel class) but instead naming things in relationship to what they are doing (their intention). 

I know we have been doing it like this and for consistency I would keep it if this is the case, but a more proper for me would be something like ```observeData()```, which is what it is really happening. 

My 2 cents here.

File: app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt [hunk 2]
```diff
@@ -5,20 +5,41 @@ import android.view.LayoutInflater
...
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         return inflater.inflate(R.layout.fragment_settings_support, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initToolbar()
+        initViewModel()
+
+        settingsSupportSupportWebsiteButton.setOnClickListener {
```
Comment (mejdoo) [inline app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt]: I know that you are respecting the guideline here but probably its better to rename this to `settingsSupportWebsiteButton`

Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt]: And move this to a private function `initSupportWebsiteButtonListener()`

File: app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt [hunk 3]
```diff
@@ -5,20 +5,41 @@ import android.view.LayoutInflater
...
+            settingsSupportViewModel.onSupportWebsiteClicked()
+        }
+        settingsSupportContactButton.setOnClickListener {
```
Comment (mejdoo) [inline app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt]: same here initContactButtonListener()

**Comment by mejdoo** [comment]
👍 Great Job! I left some minor comments.

File: app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt (hunk 1 above)
Comment (BradleyWire) [reply app/src/main/kotlin/com/waz/zclient/settings/support/SettingsSupportFragment.kt]: When we do a full retrospective of this feature at the end, we can look at tweaking these types of issues.

**Pull request merged**